### PR TITLE
Define v0.3 of the link predicate

### PR DESCRIPTION
**Pull request by adityasaky**
* Updates materials to use ResourceDescriptor
* Introduces a protobuf definition for the link predicate.

### CODE REVIEW

**Comment by adityasaky** [comment]
@SantiagoTorres

File: spec/predicates/link.md [hunk 1]
```diff
@@ -1,76 +1,102 @@
...
 
-## TODO
-
--   [ ] Bump up the in-toto version from 0.9 to 1.0 once
-    [in-toto/docs issue #46](https://github.com/in-toto/docs/issues/46) is
-    resolved.
-
 ## Version History
 
+-   0.3: Updated `materials` to use `ResourceDescriptor`.
```
Comment (joshuagl) [inline spec/predicates/link.md:96]: materials seems to have changed from an object to a list of `ResourceDescriptor`, is that right?

also:
* changed TypeURI
* command changed from string to list

Comment (adityasaky) [reply spec/predicates/link.md:96]: Good point. I should note that command is currently a string, it was misrepresented in v0.2.

Comment (adityasaky) [reply spec/predicates/link.md]: Updated the changelog.

**Review by joshuagl — APPROVED** [review]

**Review by marcelamelara — COMMENTED** [review]
Thank you @adityasaky. I only have a couple minor changes.

File: spec/predicates/link.md [hunk 2]
```diff
@@ -1,76 +1,104 @@
...
   "subject": [{ ... }],
 
   // Predicate:
-  "predicateType": "https://in-toto.io/Link/v0.2",
+  "predicateType": "https://in-toto.io/attestation/link/v0.3",
   "predicate": {
     "name": "...",
-    "command": "...",
-    "materials": { ... },
+    "command": [ ... ],
```
[the remark this answers is not in the document]
Comment (marcelamelara) [reply spec/predicates/link.md:47]: Ah, so each string in the list is an argument... I'd still clarify that int he schema I think.

[the remark this answers is not in the document]
Comment (adityasaky) [reply spec/predicates/link.md:47]: Should we copy the schema in from the in-toto spec? It just currently points there and highlights differences.

**Review by marcelamelara — APPROVED** [review]
Thanks for the updates, LGTM!

**Pull request merged**